lib/lexer.mll

```diff
@@ -1,301 +0,0 @@
-(** The lexer that will feed into the parser. An OCamllex file. *)
-
-{
-  open Lexing
-  open Debug
-
-(* Boilerplate for getting line numbers for errors *)
-  let incr_linenum lexbuf =
-    let pos = lexbuf.lex_curr_p in
-    lexbuf.lex_curr_p <- { pos with
-      pos_lnum = pos.pos_lnum + 1;
-      pos_bol = pos.pos_cnum;
-    }
-
-(* Some definitions for handling includes correctly *)
-  let dup_exists l =
-    let rec dup_consecutive = function
-      | [] | [_] -> false
-      | x1 :: x2 :: tl -> x1 = x2 || dup_consecutive (x2 :: tl)
-    in
-    dup_consecutive (List.sort String.compare l)    
-
-  let include_stack = Stack.create ()
-
-  let include_paths : string list ref = ref []
-
-  let rec try_open_in paths fname =
-  match paths with
-    | [] -> raise (Errors.SyntaxError
-              (Includes ("Could not find include file \"" ^ fname ^
-                         "\" in specified include paths.\n", lexeme_end_p
-                (Stack.top include_stack))))
-    | path :: rest_of_paths ->
-    try
-      let old_path = (Stack.top include_stack).lex_start_p.pos_fname in
-      let full_path = path ^ "/" ^ fname in
-        open_in full_path, full_path ^ "\" included from \"" ^ old_path
-    with _ -> try_open_in rest_of_paths fname
-  (* TODO: Put in precise location with line number and position in the whole chain of included froms? *)
-  
-  let maybe_remove_quotes str =
-    let open Core_kernel.String in
-    if is_prefix str ~prefix:"\"" &&
-       is_suffix str ~suffix:"\""
-    then drop_suffix (drop_prefix str 1) 1
-    else str
-}
-
-(* Some auxiliary definition for variables and constants *)
-let string_literal = '"' [^'"']* '"' (* TODO: We should probably expand the alphabet *)
-let identifier = ['a'-'z' 'A'-'Z' '_'] ['a'-'z' 'A'-'Z' '0'-'9' '_']* 
-
-let integer_constant =  ['0'-'9']+
-
-let exp_literal = ['e' 'E'] ['+' '-']? integer_constant
-let real_constant1 = integer_constant? '.' ['0'-'9']* exp_literal? 
-let real_constant2 = '.' ['0'-'9']+ exp_literal?
-let real_constant3 = integer_constant exp_literal 
-let real_constant = real_constant1 | real_constant2 | real_constant3
-
-rule token = parse
-(* White space, line numers and comments *)
-    '\n'                      { lexer_logger "newline" ;
-                                incr_linenum lexbuf ; token lexbuf }
-  | [' ' '\t' '\012' '\r']    { lexer_logger "space" ;
```

Comment:
might be worth setting up a separate whitespace definition

Reply:
Done.